lib/secure-boot-lib-ami.robot

```diff
@@ -0,0 +1,228 @@
+    ${sb_menu}=    Get Slice From List    ${sb_menu}    1
+    Set Option State    ${sb_menu}    Secure Boot *\[    Disabled
+
+Enroll DB Signature
+    [Documentation]    Enroll new DB Signature. Should be called in
+    ...    Key Management Menu
+    # robocop: disable=unused-argument
+    [Arguments]    ${key_menu}    ${volume}    ${file}    @{}
+    ...    ${fail_on_duplicate}=${FALSE}
+    # robocop: enable
+    Enter Submenu From Snapshot    ${key_menu}    > Authorized Signatures
+    Select Ami Option    Append
+    Select Ami Option    ${FALSE}
+    Select Ami Option    USB
+    Select Ami Option    ${file}
+    Select Ami Option    Public Key Certificate
+    ${status}=    Read From Terminal Until    ---/
+    Should Not Contain    ${status}    Load Error
+    Press Enter
+    Read From Terminal Until    ---/
+    ${status}=    Read From Terminal Until    ---/
+    IF    ${fail_on_duplicate} == ${TRUE}
+        Should Contain    ${status}    Success
+    ELSE
+        Should Contain Any    ${status}    Success    Already started
+    END
+    Press Enter
+
+Enter UEFI Shell
+    [Documentation]    Boots into efi shell
+    Enter Setup Menu And Return Construction
+    Press Key N Times    1    ${ARROW_LEFT}
+    Boot Option    EFI Shell
+    Read From Terminal Until    Shell>
+
+Delete Boot Option Ami
+    [Documentation]    Delete boot option. Should be called from Boot Menu.
+    ...    At the end returns to the top of Boot menu
+    [Arguments]    ${boot_option}    ${fail_if_doesnt_exist}=${FALSE}
+    Press Key N Times And Enter    1    ${END}
+    Read From Terminal
+    Press Enter
+    ${status}=    Run Keyword And Return Status    Select Ami Option    ${boot_option}
+    Press Key N Times    2    ${ESC}
+    IF    ${status} == ${FALSE}
+        IF    ${fail_if_doesnt_exist} == ${TRUE}
+            Fail    Couldn't delete boot menu
+        END
+    END
+
+Add Boot Option Ami
+    [Documentation]    Add new boot option. Should be called from Main Menu.
+    ...    If disk is empty then chooses first disk in menu. At the end returns
+    ...    to the top of Boot menu
+    [Arguments]    ${file}    ${boot_option}    ${volume}
+    Press Key N Times    2    ${ARROW_LEFT}
+    Delete Boot Option Ami    ${boot_option}
+    Press Key N Times    1    ${END}
+    Read From Terminal
+    Press Key N Times And Enter    1    ${ARROW_UP}
+    Read From Terminal Until    Add New Boot Option
+    Press Enter
+    Write Bare Into Terminal    ${boot_option}
+    Press Enter
+    Read From Terminal
+    Press Key N Times And Enter    1    ${ARROW_DOWN}
+    Select Ami Option    ${volume}
+    Select Ami Option    ${file}
+    Press Key N Times And Enter    1    ${ARROW_DOWN}
+    Press Enter
+    Press Key N Times    1    ${ESC}
+    Press Key N Times    1    ${HOME}
+
+Boot Efi File And Return Response
+    [Documentation]    Boots EFI file and returns response.
+    ...    If read_only_first_line is true then reads only up to \r\n else
```

Comment:
@m-iwanicki could you also help here? Do you have any idea how this could be implemented better?

Reply:
@TomaszAIR @macpijan When booting hello_world system prints for very short time "Hello World" after whih it returns to Setup Menu. telnet reads for a very short time: "Hello World\r\n" and then reads Setup Menu which either clears screen or just overwrites this Hello World.
End result is that in output text from Read From Terminal there is no "Hello World" line.
This keyword is also used when trying to boot unsigned EFI file in which case it will return Error Dialog (that ends in `---/`.